### DISCUSSION TITLE
Support an "auto" value for the current HVAC mode?

### DISCUSSION THREAD

**Discussion by Rjevski** [home-assistant/architecture#673] [Entity Models] [answered]
While working on a climate integration I encountered an issue with how to map the current HVAC action to Home Assistant. The unit can be configured to auto mode where it will automatically maintain a temperature range, cooling or heating as necessary. The actual range isn't known by Home Assistant (and this integration is actually for a gateway compatible with multiple different HVACs, so even hardcoding the range wouldn't work as it'll be manufacturer-dependent) and the action (from a refrigeration point of view, whether it's cooling or heating) is unknown either.

Now you could indeed do tricks based on current & target temperature (`if current > target then cooling else heating`) but it seems like at least HomeKit has the concept of reporting "AUTO" as a valid HVAC status: https://developer.apple.com/documentation/homekit/hmcharacteristictypecurrentheatingcooling so I was wondering if we can support this in Home Assistant? This means that for units where the state is not precisely queryable you can just return "AUTO" instead of trying to infer based on temperatures.

I expect the following work needs to be done:

* backend code change ([draft here](https://github.com/home-assistant/core/pull/60262))
* update documentation
* update any integrations that use `climate.hvac_action` to be able to deal with the new value - HomeKit is taken care of, are there other "meta-integrations" that looks at the state of other integrations?
* update translations
* update frontend

Another question is whether it's justified to do this work now. I had this issue with the Coolmaster integration but looking at the usages of `CURRENT_HVAC_COOL` it seems like other integrations don't have this particular problem (or maybe they're just incorrectly mapping it and I haven't looked too closely) so I'm keen to hear others' opinions as to whether this is worth the trouble.

**Comment by Jc2k** [comment]
The current API in HA works fine for homekit_controller. I think that webpage is confusing maybe? According to the latest device implementers spec, homekit devices can't report their current heating/cooling status as auto, only heating/cooling/off.

They can report their current target as auto.

**Reply by Jc2k** [reply]
BTW, homekit_controller maps homekit auto to heat_cool.

**Comment by MartinHjelmare** [comment, marked as answer]
I don't see the value of this feature. If we don't know the current HVAC action we return None for that property. Auto is not a current action, it's the mode of the HVAC.
